engine/baml-lib/jsonish/src/deserializer/coercer/coerce_literal.rs

```diff
@@ -46,6 +47,22 @@ impl TypeCoercer for LiteralValue {
             Some(v) => v,
         };
 
+        // If we get an object with a single key-value pair, try to extract the value
+        if let jsonish::Value::Object(obj) = value {
+            if obj.len() == 1 {
+                let (key, inner_value) = obj.iter().next().unwrap();
+                // only extract value if it's a primitive (not an object or array, hoping to god its fixed)
+                match inner_value {
+                    jsonish::Value::Number(_) | jsonish::Value::Boolean(_) | jsonish::Value::String(_) => {
+                        let mut result = self.coerce(ctx, target, Some(inner_value))?;
+                        result.add_flag(Flag::ImpliedKey(key.clone()));
+                        return Ok(result);
```

Comment:
we should use a different flag here: `ObjectToPrimitive`  instead.

implied key actually means the key doesn't exist, but we were able to assume it does. Sorry no docs about that.